exporters/jaeger/src/recordable.cc

```diff
@@ -39,14 +56,97 @@ void JaegerRecordable::PopulateAttribute(nostd::string_view key,
   }
-  // TODO: extend other AttributeType to the types supported by Jaeger.
+  else if (nostd::holds_alternative<nostd::span<const int>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const int>>(value))
+    {
+      AddTag(std::string{key}, int64_t{val}, tags);
+    }
+  }
+  else if (nostd::holds_alternative<nostd::span<const uint8_t>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const uint8_t>>(value))
+    {
+      AddTag(std::string{key}, int64_t{val}, tags);
+    }
+  }
+  else if (nostd::holds_alternative<nostd::span<const bool>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const bool>>(value))
+    {
+      AddTag(std::string{key}, val, tags);
+    }
+  }
+  else if (nostd::holds_alternative<nostd::span<const int>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const int>>(value))
+    {
+      AddTag(std::string{key}, int64_t{val}, tags);
+    }
+  }
+  else if (nostd::holds_alternative<nostd::span<const int64_t>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const int64_t>>(value))
+    {
+      AddTag(std::string{key}, val, tags);
+    }
+  }
+  else if (nostd::holds_alternative<nostd::span<const unsigned int>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const unsigned int>>(value))
+    {
+      AddTag(std::string{key}, int64_t{val}, tags);
+    }
+  }
+  else if (nostd::holds_alternative<nostd::span<const uint64_t>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const uint64_t>>(value))
+    {
+      AddTag(std::string{key}, std::to_string(val), tags);
+    }
+  }
+  else if (nostd::holds_alternative<nostd::span<const double>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const double>>(value))
+    {
+      AddTag(std::string{key}, val, tags);
+    }
+  }
+  else if (nostd::holds_alternative<nostd::span<const nostd::string_view>>(value))
+  {
+    for (const auto &val : nostd::get<nostd::span<const nostd::string_view>>(value))
+    {
+      AddTag(std::string{key}, std::string{val}, tags);
+    }
+  }
+  else
+  {
+    OTEL_INTERNAL_LOG_ERROR(
+        "[TRACE JAEGER Exporter] SetAttribute() failed, attribute type not supported ");
+  }
 }
 
 void JaegerRecordable::PopulateAttribute(nostd::string_view key,
                                          const sdk::common::OwnedAttributeValue &value,
                                          std::vector<thrift::Tag> &tags)
 {
-  if (nostd::holds_alternative<int64_t>(value))
+  if (nostd::holds_alternative<int>(value))
+  {
+    AddTag(std::string{key}, int64_t{nostd::get<int>(value)}, tags);
+  }
+  else if (nostd::holds_alternative<int32_t>(value))
+  {
+    AddTag(std::string{key}, int64_t{nostd::get<int32_t>(value)}, tags);
+  }
+  else if (nostd::holds_alternative<uint32_t>(value))
+  {
+    AddTag(std::string{key}, int64_t{nostd::get<uint32_t>(value)}, tags);
+  }
+  else if (nostd::holds_alternative<uint64_t>(value))
```

Comment:
Curious - is there specific ordering? (I saw uint32 was after int32, while uint64 was before int64, hence the ask).

Reply:
I made it consistent. The order should not matter though.

Reply:
Thanks!